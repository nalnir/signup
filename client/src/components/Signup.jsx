import React from 'react';
import { hot } from 'react-hot-loader';
import {
  Input,
  InputLabel,
  FormControl,
  InputAdornment,
} from '@material-ui/core';
import Email from '@material-ui/icons/Email';
import VpnKey from '@material-ui/icons/VpnKey';


function Signup() {
  return (
    <div>
      <div>
        <FormControl required>
          <InputLabel htmlFor="signup-first-name" className="first-name">First Name</InputLabel>
          <Input type="text" id="signup-first-name" className="first-name" placeholder="First Name" />
        </FormControl>
      </div>
      <div>
        <FormControl required>
          <InputLabel htmlFor="signup-last-name" className="last-name">Last Name</InputLabel>
          <Input type="text" id="signup-last-name" className="last-name" placeholder="Last Name" />
        </FormControl>
      </div>
      <div>
        <FormControl required>
          <InputLabel htmlFor="signup-email-address" className="email-address">Email</InputLabel>
          <Input
            type="email"
            id="signup-email-address"
            className="email-address"
            placeholder="drone@globex.com"
            startAdornment={
              (
                <InputAdornment position="start">
                  <Email />
                </InputAdornment>
              )
            }
          />
        </FormControl>
      </div>
      <div>
        <FormControl required>
          <InputLabel htmlFor="signup-password" className="password">Password</InputLabel>
          <Input
            type="email"
            id="signup-password"
            className="password"
            startAdornment={
              (
                <InputAdornment position="start">
                  <VpnKey />
                </InputAdornment>
              )
            }
          />
        </FormControl>
      </div>
    </div>
  );
}


export default hot(module)(Signup);
