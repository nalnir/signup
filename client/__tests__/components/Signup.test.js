import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Signup from '../../src/components/Signup.jsx';
import setupJestEnzyme from '../setup/setupJestEnzyme.js';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import Input from '@material-ui/core/Input';
import Button from '@material-ui/core/Button';


describe('signup', () => {
  var signup;

  beforeEach( () => {
    signup = shallow(<Signup />);
  });

  it('renders signup correctly', () => {
    expect(signup).toMatchSnapshot();
  });

  it('has a place to enter the first and last name', () => {
    expect(signup.find(Input).filter('.first-name')).toHaveLength(1);
    expect(signup.find(Input).filter('.last-name')).toHaveLength(1);
  });

  it('indicates which field to enter the first and last name', () => {
    expect(signup.find(InputLabel).filter('.first-name')).toHaveLength(1);
    expect(signup.find(InputLabel).filter('.last-name')).toHaveLength(1);
  });

  it('has a place to enter the email address', () => {
    expect(signup.find(Input).filter('.email-address')).toHaveLength(1);
  });

  it('has a place to enter the password', () => {
    expect(signup.find(Input).filter('.password')).toHaveLength(1);
  });

  it('has a way to submit information for signup', () => {
    expect(signup.find(Button).filter('.create-new-account')).toHaveLength(1);
  });
});