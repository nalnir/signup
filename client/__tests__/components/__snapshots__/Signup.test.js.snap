// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup renders signup correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <label
            className="first-name"
            htmlFor="signup-first-name"
          >
            First Name 
          </label>
          <input
            className="first-name"
            id="signup-first-name"
            type="text"
          />
        </div>,
        <div>
          <label
            className="last-name"
            htmlFor="signup-last-name"
          >
            Last Name 
          </label>
          <input
            className="last-name"
            id="signup-last-name"
            type="text"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="first-name"
              htmlFor="signup-first-name"
            >
              First Name 
            </label>,
            <input
              className="first-name"
              id="signup-first-name"
              type="text"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "First Name ",
              "className": "first-name",
              "htmlFor": "signup-first-name",
            },
            "ref": null,
            "rendered": "First Name ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "first-name",
              "id": "signup-first-name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="last-name"
              htmlFor="signup-last-name"
            >
              Last Name 
            </label>,
            <input
              className="last-name"
              id="signup-last-name"
              type="text"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Last Name ",
              "className": "last-name",
              "htmlFor": "signup-last-name",
            },
            "ref": null,
            "rendered": "Last Name ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "last-name",
              "id": "signup-last-name",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <label
              className="first-name"
              htmlFor="signup-first-name"
            >
              First Name 
            </label>
            <input
              className="first-name"
              id="signup-first-name"
              type="text"
            />
          </div>,
          <div>
            <label
              className="last-name"
              htmlFor="signup-last-name"
            >
              Last Name 
            </label>
            <input
              className="last-name"
              id="signup-last-name"
              type="text"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="first-name"
                htmlFor="signup-first-name"
              >
                First Name 
              </label>,
              <input
                className="first-name"
                id="signup-first-name"
                type="text"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "First Name ",
                "className": "first-name",
                "htmlFor": "signup-first-name",
              },
              "ref": null,
              "rendered": "First Name ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "first-name",
                "id": "signup-first-name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="last-name"
                htmlFor="signup-last-name"
              >
                Last Name 
              </label>,
              <input
                className="last-name"
                id="signup-last-name"
                type="text"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Last Name ",
                "className": "last-name",
                "htmlFor": "signup-last-name",
              },
              "ref": null,
              "rendered": "Last Name ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "last-name",
                "id": "signup-last-name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
